src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -339,10 +362,22 @@ private String locateNoCheck(String name) {
      */
     @Nullable
     private String locate(String name) {
-      String path = locateNoCheck(name);
-      if (modulePaths.contains(path)) {
-        return path;
+      InputPathData pathData = new InputPathData(canonicalizePath(name));
+      InputPathData modulePath = modulePaths.get(pathData.ensureLeadingSlash());
+      if (modulePath != null) {
```

Comment:
Mainly because of the great difficulty of landing anything that deals with the deps package. I did my absolute best to change the bare minimum of test cases to force the functionality to adhere to existing results.

My preferences is to return the path as absolute here if you're willing to try it.